crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -456,24 +455,22 @@ async fn c8y_mapper_alarm_with_custom_fragment_mapping_to_c8y_json() {
 
     broker
         .publish_with_opts(
-            "tedge/alarms/major/temperature_alarm",
+            "tedge/alarms/major/custom_temperature_alarm",
             r#"{ "text": "Temperature high","time":"2023-01-25T18:41:14.776170774Z","customFragment": {"nested":{"value": "extra info"}} }"#,
             mqtt_channel::QoS::AtLeastOnce,
             true,
         )
         .await
         .unwrap();
 
-    let expected_msg = json!({"severity":"MAJOR","type":"temperature_alarm","time":"2023-01-25T18:41:14.776170774Z","text":"Temperature high","customFragment":{"nested":{"value":"extra info"}}});
-
-    while let Ok(Some(msg)) = messages.next().with_timeout(TEST_TIMEOUT_MS).await {
```

Comment:
So the sporadically test failure is fixed by replacing this block by `mqtt_tests::assert_received_all_expected`? It looks all other tests have this replacement.

I am just curious, because other locations than alarm also use this `while let` block for testing.

Reply:
Yup, using `mqtt_tests::assert_received_all_expected` fixed the issue. Before I was trying to assert the json objects and was relying on the `timeout` error when it does not receive. But, the call `with_timeout` does not return an error, and because of this, the tests were always passing.
I did check at least two other places where `while let` is being used, felt it's okay.